azkaban-executor/azkaban-emr-jobtype/src/main/java/uk/gov/dwp/dataworks/azkaban/services/DependencyService.java

```diff
@@ -13,137 +11,52 @@
+        this.dataProductStatusService = dataProductStatusService;
         this.metadataTableName = metadataTableName;
         this.exportDate = exportDate;
     }
 
-    private static Map<String, AttributeValue> primaryKey(final Map<String, AttributeValue> item) {
-        return item.entrySet().stream().filter(entry -> entry.getKey().equals(CORRELATION_ID_FIELD) || entry.getKey()
-                                                                                                            .equals(DATA_PRODUCT_FIELD))
-                   .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-    }
-
-    private static Map<String, AttributeValue> valueMap(final String product, final String exportDate) {
-        Map<String, AttributeValue> valueMap = new HashMap<>();
-        valueMap.put(":export_date", new AttributeValue().withS(exportDate));
-        valueMap.put(":product", new AttributeValue().withS(product));
-        return valueMap;
-    }
-
-    private static Map<String, String> nameMap() {
-        Map<String, String> nameMap = new HashMap<>();
-        nameMap.put("#export_date", DATE_FIELD);
-        nameMap.put("#product", DATA_PRODUCT_FIELD);
-        return nameMap;
-    }
-
-    private static boolean hasFinished(final String status) {
-        return hasSucceeded(status) || hasFailed(status);
-    }
-
-    private static boolean hasFailed(final String status) {
-        return hasStatus(FAILED_COMPLETION_STATUS, status);
-    }
-
-    private static boolean hasSucceeded(final String status) {
-        return hasStatus(SUCCESSFUL_COMPLETION_STATUS, status);
-    }
-
-    private static boolean hasStatus(final String required, final String status) {
-        return required.equalsIgnoreCase(status);
-    }
-
-    private static String itemString(final Map<String, AttributeValue> item) {
-        return item.get(CORRELATION_ID_FIELD).getS() + "/" + item.get(DATA_PRODUCT_FIELD).getS() + "/" + item
-                .get(DATE_FIELD).getS();
-    }
-
-    private static int pollIntervalMilliseconds() {
-        return Integer.parseInt(System.getProperty("poll.interval.milliseconds", "10000"));
-    }
-
-    private static int pollTimeoutMilliseconds() {
-        return Integer.parseInt(System.getProperty("poll.timeout.milliseconds", "3600000"));
-    }
-
-    public Optional<Map<String, AttributeValue>> successfulDependency(final String product) {
+    public Optional<Map<String, AttributeValue>> successfulDependency(final String product,
+            final String... collections) {
         return proceed.get() ?
-                dependencyMetadata(product).filter(this::dependencySucceeded)
-                                           .flatMap(x -> dependencyMetadata(product)) :
+                dependencyMetadata(product).filter(collections.length > 0 ?
+                        item -> collectionStatusService
+                                .collectionsSucceeded(item.get(CORRELATION_ID_FIELD).getS(), collections) :
```

Comment:
Same question for `CORRELATION_ID_FIELD`.

Reply:
In the MetadataService interface